catboost/python-package/catboost/core.py

```diff
@@ -1108,6 +1107,12 @@ def _fit(self, X, y, cat_features, pairs, sample_weight, group_id, group_weight,
              column_description, verbose_eval, metric_period, silent, early_stopping_rounds,
              save_snapshot, snapshot_file, snapshot_interval):
 
+        if X is None:
+            raise CatBoostError("X must not be None")
+
+        if y is None and not isinstance(X, Pool):
```

Comment:
X could also be a string with dataset file path

Reply:
@kizill thanks, I saw that but on the documentation of `y` it is not described:
``` y : list or numpy.array or pandas.DataFrame or pandas.Series, optional (default=None)
            Labels, 1 dimensional array like.
            Use only if X is not catboost.Pool.```

If I read this, I believe that  whenever y is None then X should be an instance of Pool. I think the doc shoudl say that. Should I update the docstring as well as the `isinstance` condition?

Reply:
@kizill I have added the additional condition.